packages/react/src/primitives/TextAreaField/TextAreaField.tsx

```diff
@@ -0,0 +1,76 @@
+import * as React from 'react';
+import classNames from 'classnames';
+
+import { ComponentClassNames } from '../shared/constants';
+import { FieldDescription, FieldErrorMessage } from '../Field';
+import { Flex } from '../Flex';
+import { Label } from '../Label';
+import { Primitive } from '../types';
+import { splitPrimitiveProps } from '../shared/styleUtils';
+import { TextArea } from '../TextArea';
+import { TextAreaFieldProps } from '../types/textAreaField';
+import { useStableId } from '../shared/utils';
+
+export const DEFAULT_ROW_COUNT = 3;
+
+const TextAreaFieldPrimitive: Primitive<TextAreaFieldProps, 'textarea'> = (
```

Comment:
Splitting out makes type here clearer!

Reply:
Definitely, it's going to be a huge improvement!